docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,370 @@
+    "silence",  
+    "unknown",  
+    "yes",  
+    "no",  
+    };
+    ```
 
-To save time, we will skip this step and instead download the [tiny_conv.tflite](https://developer.arm.com/-/media/Files/downloads/Machine%20learning%20how-to%20guides/tiny_conv.tflite?revision=495eb362-4325-49b8-b3ba-3141df0c9b95&la=en&hash=0F37BA2C5DE95A1561979CDD18973171767A47C3).
+    The code uses this array to map the output of the model to the correct value. Because we specified our wanted_words<!--is that underscore correct?--> as “up, down” in the training script, we should update this array to reflect these words in the same order.
 
-The final step in the process is to convert this model into a C file that we can drop into our Mbed project.
+1. Edit the code so that "up" replaces "yes" and "down" replaces "no":
 
-To do this conversion, we will use a tool called xxd. Issue the following command:
+    ```
+    const char* kCategoryLabels[kCategoryCount] = {  
+    "silence",  
+    "unknown",  
+    "up",  
+    "down",  
+    };
+    ```
 
-```xxd -i  tiny_conv.tflite > ../micro_features/model.cc```
+1. Update the code in `command_responder.cc` to reflect these new labels, modifying the if statements and the DisplayStringAt call:
 
-Next, we need to update model.cc so that it is compatible with our code. First, open the file. The top two lines should look similar to the following code, although the exact variable name and hex values may be different:
-```
-const  unsigned  char  g_model[] DATA_ALIGN_ATTRIBUTE = {  
-0x18, 0x00, 0x00, 0x00, 0x54, 0x46, 0x4c, 0x33, 0x00, 0x00, 0x0e, 0x00,
-```
-You need to add the include from the following snippet and change the variable declaration without changing the hex values:
-```
-#include "tensorflow/lite/micro/examples/micro_speech/micro_features/model.h"  
-const unsigned char g_tiny_conv_micro_features_model_data[] = {  
-0x18, 0x00, 0x00, 0x00, 0x54, 0x46, 0x4c, 0x33, 0x00, 0x00, 0x0e, 0x00,
-```
-Next, go to the very bottom of the file and find the unsigned int variable.
-```
-unsigned int tiny_conv_tflite_len = 18216;
-```
-Change the declaration to the following code, but do not change the number assigned to it, even if your number is different from the one in this guide.
-```
-const int g_tiny_conv_micro_features_model_data_len = 18216;
-```
-Finally, save the file, then copy the ```tiny_conv_micro_features_model_data.cc```file into the ```tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/tensorflow/lite/micro/examples/micro_speech/micro_features``` directory.
+    ```
+    void RespondToCommand(tflite::ErrorReporter* error_reporter,  
+    int32_t current_time, const char* found_command,  
+    uint8_t score, bool is_new_command) {  
+        if (is_new_command) {  
+            error_reporter->Report("Heard %s (%d) @%dms", found_command, score,  
+            current_time);  
+        if(strcmp(found_command, "up") == 0) {  
+            lcd. Clear(0xFF0F9D58);  
```

Comment:
Where did that LCD screen come from 😄?

Reply:
This is a mistake. This board does not have an LCD screen and the command_responder.cc should not be changed if the model is re-trained.